src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfo.cs

```diff
@@ -127,7 +127,7 @@ private void DetermineIgnoreCondition(JsonIgnoreCondition? ignoreCondition)
         {
             if (ignoreCondition != null)
             {
-                Debug.Assert(PropertyInfo != null);
+                Debug.Assert(MemberInfo != null);
```

Comment:
We need to add the `IgnoreReadonlyFields` property specified in https://github.com/dotnet/runtime/issues/876#issue-537934155 and update `DetermineSerializationCapabilities()` above accordingly.